### PR TITLE
mariadb: 10.2.9 -> 10.2.11

### PR DESCRIPTION
**Pull request by Izorkin**
###### Motivation for this change
Update version. Fix cicle error.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Review by Izorkin — COMMENTED** [review]
poco: not support MySQL with mysql.connector-c

**Review by Izorkin — COMMENTED** [review]
dspam: not build with mariadb

**Review by Izorkin — COMMENTED** [review]
postfix: build error with mysql.connector-c

**Review by Izorkin — COMMENTED** [review]
libmysql: no more apps support build with mariadb connector-c

**Review by Izorkin — COMMENTED** [review]
amarok: build error with mariadb 10.2

**Review by Izorkin — COMMENTED** [review]
qt-4: error build with mariadb connector-c

**Comment by globin** [comment]
I've rebased #30546 and (partly) cherry-picked some your changes, can you please rebase on top of that again.

**Review by ttuegel — APPROVED** [review]
For my part, the changes to Akonadi look fine.

File: pkgs/applications/kde/akonadi/default.nix [hunk 1]
```diff
@@ -20,10 +23,10 @@ mkDerivation {
...
   outputs = [ "out" "dev" ];
   NIX_CFLAGS_COMPILE = [
-    ''-DNIXPKGS_MYSQL_MYSQLD="${lib.getBin mysql}/bin/mysqld"''
-    ''-DNIXPKGS_MYSQL_MYSQLADMIN="${lib.getBin mysql}/bin/mysqladmin"''
-    ''-DNIXPKGS_MYSQL_MYSQL_INSTALL_DB="${lib.getBin mysql}/bin/mysql_install_db"''
-    ''-DNIXPKGS_MYSQL_MYSQLCHECK="${lib.getBin mysql}/bin/mysqlcheck"''
+    ''-DNIXPKGS_MYSQL_MYSQLD="${mysql}/bin/mysqld"''
+    ''-DNIXPKGS_MYSQL_MYSQLADMIN="${mysql}/bin/mysqladmin"''
+    ''-DNIXPKGS_MYSQL_MYSQL_INSTALL_DB="${mysql}/bin/mysql_install_db"''
+    ''-DNIXPKGS_MYSQL_MYSQLCHECK="${mysql}/bin/mysqlcheck"''
```
Comment (globin) [inline pkgs/applications/kde/akonadi/default.nix]: I won't include this one as it might very well be that we add a bin output to mysql in the future.

Comment (Izorkin) [reply pkgs/applications/kde/akonadi/default.nix]: ok, rebased

**Comment by globin** [comment]
This is now all included in the parent PR (although github is having difficulties in displaying this correctly, I've just double-checked the diff), thank you very much for your work on this!

**Pull request closed without merging**